### DISCUSSION TITLE
How to use IRoleAssignment

### DISCUSSION THREAD

**Discussion by Ofer-Gal** [pnp/pnpcore#904] [Q&A]
I know how to get the visitors group
`ISharePointGroup sharePointGroup = await web.SiteGroups.FirstOrDefault(p => p.Title.Contains("Visitors")).GetAsync();
`
And a reference to a list.
`IList DPCDocumentsList = await pnpContext.Web.Lists.GetByTitleAsync("DPC Documents");`
Now I need to revoke "contribute" and grant "Read" to the Visitors on the "DPC Documents" library.
How do I do it? 
Thank you Much

**Comment by jansenbe** [comment]
@Ofer-Gal : see https://pnp.github.io/pnpcore/using-the-sdk/security-intro.html#assigning-roles.

You would need to first break permission inheritance on your list (https://pnp.github.io/pnpcore/using-the-sdk/security-intro.html#breaking-permission-inheritance), set copyRoleAssignments to false so you start clean. Next you add the Read role to the Visitors group for that list (https://pnp.github.io/pnpcore/using-the-sdk/security-intro.html#assigning-roles-to-a-user-or-group). The documentation shows this for a user, but you can do the same with the `sharePointGroup` reference you've already loaded.

**Reply by Ofer-Gal** [reply]
Thank you very much.
In my case, the permission inheritance is already broken at the site provisioning. 
Your examples are for the list. I need to do it on the file or item. 
I will try to do the same to the item.
Tried. works fine! Thanks
